### DISCUSSION TITLE
Running a what-if command from azd ?

### DISCUSSION THREAD

**Discussion by xaviermignot** [Azure/azure-dev#129] [General]
I use Terraform a lot and run `terraform plan` all the time to check my code and foresee my changes before actually applying them.
I'm just getting started with Bicep and found an equivalent to `terraform plan` with `what-if`commands. I can't see a way to run `what-if` commands through azd, from my understanding it only runs `create` commands.

The only way I have found is to run a command like this from the `infra` folder:  
`az deployment sub what-if -l westeurope --template-file main.bicep -p ../.azure/dev/main.parameters.json`
But it looks like a workaround, right ? 😅 

So is there or will there be a way to foresee my changes on the Azure resources using azd ?

**Comment by savannahostrowski** [comment]
Thanks for the question! We currently don't have support for any similar command in `azd`. That being said, this is a solid request so I've made some issues to track this request. 

I've created two issues to track this:
https://github.com/Azure/azure-dev/issues/141 (diff support)
https://github.com/Azure/azure-dev/issues/142 (making sure that we validate upfront that a developer has sufficient perms to run a template)

**Comment by savannahostrowski** [comment]
Closing this thread in favour of those two issues!
